src/reducers/polling.js

```diff
@@ -0,0 +1,325 @@
+          type: `poll/${prefix}_SENT`,
+          payload: { txHash: tx.hash }
+        });
+      },
+      mined: _ => {
+        dispatch({ type: `poll/${prefix}_SUCCESS` });
+        resolve();
+      },
+      error: (_, err) => {
+        dispatch({ type: `poll/${prefix}_FAILURE`, payload: err });
+        dispatch(addToastWithTimeout(ToastTypes.ERROR, err));
+        resolve();
+      }
+    });
+  });
+
+export const pollsRequest = () => ({
+  type: POLLS_REQUEST
+});
+export const pollsSuccess = polls => ({
+  type: POLLS_SUCCESS,
+  payload: polls
+});
+export const pollsFailure = () => ({
+  type: POLLS_FAILURE
+});
+
+export const setOptionVotingFor = (pollId, optionId) => ({
+  type: POLLS_SET_OPTION_VOTING_FOR,
+  payload: { pollId, optionId }
+});
+
+// Writes ---
+
+export const voteForPoll = (pollId, optionId) => async dispatch => {
+  dispatch({ type: POLL_VOTE_REQUEST });
+
+  const pollVote = window.maker.service('govPolling').vote(pollId, optionId);
+  await handleTx({
+    txObject: pollVote,
+    prefix: 'VOTE',
+    dispatch
+  });
+  dispatch(setOptionVotingFor(pollId, optionId));
+};
+
+export const withdrawVoteForPoll = pollId => async dispatch => {
+  dispatch({ type: POLL_VOTE_REQUEST });
+
+  const pollVote = window.maker.service('govPolling').vote(pollId, 0);
+  await handleTx({
+    txObject: pollVote,
+    prefix: 'VOTE',
+    dispatch
+  });
+
+  dispatch(setOptionVotingFor(pollId, 0));
+};
+
+// Reads ---
+
+const getAllWhiteListedPolls = async () => {
+  const pollsList = await window.maker
+    .service('govPolling')
+    .getAllWhitelistedPolls();
+
+  const polls = uniqBy(pollsList, p => p.url);
+  console.log('unique polls', polls);
+  return polls;
+};
+
+export const getOptionVotingFor = (address, pollId) => async dispatch => {
+  const optionId = await window.maker
+    .service('govPolling')
+    .getOptionVotingFor(address, pollId);
+  dispatch(setOptionVotingFor(pollId, optionId));
+};
+
+const fetchPollFromUrl = async url => {
+  console.log('url to fetch', url);
+  const res = await fetch(url);
+  await check(res);
+  const json = await res.json();
+  if (!json['voteId']) {
+    return null;
+  } else {
+    console.log('valid poll', json);
+    return json;
+  }
+};
+
+const formatOptions = options => {
+  return Object.values(options);
+};
+
+const formatYamlToJson = data => {
+  const json = matter(data.about);
+  const { content } = json;
+  const { title, summary, options, discussion_link } = json.data;
+  return {
+    voteId: data.voteId,
+    title,
+    summary,
+    options: formatOptions(options),
+    discussion_link,
+    content,
+    rawData: data.about
+  };
+};
+
+const isPollActive = (startDate, endDate) => {
+  const now = new Date();
+  return startDate <= now && endDate > now ? true : false;
+};
+
+export const getVoteBreakdown = async (pollId, options) => {
+  // const { options: breakdownOpts } = await mockGetVoteHistory(pollId);
+
+  const voteHistory = await window.maker
```

Comment:
would it make sense to use `getMkrSupport` here? cc @tyler17

Reply:
Yeah this voteHistory function was meant for the graph of votes over time - are we not doing that graph?
I could add another simple function to the polling service called getCurrentVoteBreakdown or something that just calls getMkrSupport with `POSTGRES_MAX_INT` as the blockNumber which would do what we want.

Reply:
yea, afaict we won't have the graph for this release. And I think `getMkrSupport` should be good for us as is. Maybe we just add a direct pass through to the polling service. For active polls, we'd want the support at `POSTGRES_MAX_INT`, and for past polls we'd want support at `pollEndBlock`. is that right or do am I misunderstanding the query there?

Reply:
Well the query only counts votes that were cast while the poll was active so you should be able to just pass in max block number in both cases

Reply:
cool, i was wondering if that's how it worked. what if somebody's voting weight has changed since the poll ended?

Reply:
Oh wait you're right, forget what I said.  As the query is currently written, it will take the mkr balance at whatever block number is supplied, so your solution probably makes the most sense.